Index out of bounds on login

Got this panic while trying to test with a wildcard-ish grant:

```
grants:
  - group: Everyone
    role: cluster-admin
    resource: kubernetes
```

```
$ infra login
? Select an Infra host: localhost:443
  Logging in to localhost:443
Could not verify certificate for host "localhost:8443": Get "https://localhost:443": x509: “Infra” certificate is not trusted
? Are you sure you want to continue? Yes
? Select a login method: okta (example.okta.com)
  Logging in with okta...
  Logged in as user@example.com
panic: runtime error: index out of range [1] with length 1

goroutine 1 [running]:
github.com/infrahq/infra/internal/cmd.writeKubeconfig({0x1400079de40, 0x1, 0x1400041bef0?}, {0x14000782de0, 0x1, 0x1400025cdc0?})
        /mnt/internal/cmd/kubernetes.go:108 +0xa80
github.com/infrahq/infra/internal/cmd.updateKubeconfig(0x1400050f680, {0x1400048bbe0, 0xc})
        /mnt/internal/cmd/kubernetes.go:87 +0x2b8
github.com/infrahq/infra/internal/cmd.finishLogin({0x1400049c16f, 0xe}, {0x1400048bbe0, 0xc}, {0x14000127860, 0x18}, {0x1400041bef0, 0x23}, 0x1, 0x3ef8b2df552000)
        /mnt/internal/cmd/login.go:234 +0x3f0
github.com/infrahq/infra/internal/cmd.login({0x0, 0x0})
        /mnt/internal/cmd/login.go:177 +0x854
github.com/infrahq/infra/internal/cmd.newLoginCmd.func1(0x140008cac80?, {0x1070d60c0, 0x0, 0x0?})
        /mnt/internal/cmd/cmd.go:184 +0xb0
github.com/spf13/cobra.(*Command).execute(0x140008cac80, {0x1070d60c0, 0x0, 0x0})
        /go/pkg/mod/github.com/spf13/cobra@v1.3.0/command.go:856 +0x4c4
github.com/spf13/cobra.(*Command).ExecuteC(0x140008caa00)
        /go/pkg/mod/github.com/spf13/cobra@v1.3.0/command.go:974 +0x354
github.com/spf13/cobra.(*Command).Execute(...)
        /go/pkg/mod/github.com/spf13/cobra@v1.3.0/command.go:902
github.com/infrahq/infra/internal/cmd.Run()
        /mnt/internal/cmd/cmd.go:636 +0x3c
main.main()
        /mnt/main.go:10 +0x20
```

--- a/internal/cmd/kubernetes.go
+++ b/internal/cmd/kubernetes.go
@@ -104,6 +104,10 @@ func writeKubeconfig(destinations []api.Destination, grants []api.Grant) error {
 			continue
 		}
 
+		if len(parts) < 2 {
+			continue
+		}
+
 		cluster := parts[1]
 
 		var namespace string
@@ -120,6 +124,8 @@ func writeKubeconfig(destinations []api.Destination, grants []api.Grant) error {
 		var url, ca string
 		var exists bool
 		for _, d := range destinations {
+			// eg resource:  "kubernetes.foo.bar"
+			// eg dest name: "kubernetes.foo"
 			if strings.HasPrefix(g.Resource, d.Name) {
 				url = d.Connection.URL
 				ca = d.Connection.CA

